src/legacy/ui/public/kfetch/kfetch.ts

```diff
@@ -48,32 +47,36 @@ const interceptors: Interceptor[] = [];
 export const resetInterceptors = () => (interceptors.length = 0);
 export const addInterceptor = (interceptor: Interceptor) => interceptors.push(interceptor);
 
-export async function kfetch(
-  options: KFetchOptions,
-  { prependBasePath = true }: KFetchKibanaOptions = {}
-) {
-  const combinedOptions = withDefaultOptions(options);
-  const promise = requestInterceptors(combinedOptions).then(
-    ({ pathname, query, ...restOptions }) => {
-      const fullUrl = url.format({
-        pathname: prependBasePath ? chrome.addBasePath(pathname) : pathname,
-        query,
-      });
-
-      return window.fetch(fullUrl, restOptions).then(async res => {
-        if (!res.ok) {
-          throw new KFetchError(res, await getBodyAsJson(res));
-        }
-        const contentType = res.headers.get('content-type');
-        if (contentType && contentType.split(';')[0] === 'application/ndjson') {
-          return await getBodyAsBlob(res);
-        }
-        return await getBodyAsJson(res);
-      });
-    }
-  );
+export function createKfetch(http: HttpSetup) {
```

Comment:
does it mean that `kfetch` is deprecated and we aren't going to replicate its functionality?

Reply:
`kfetch` is a legacy concept. We are going to try and replicate as much of it in the more standard `http.fetch` world as possible.